tests/v2/validation/charts/webhook_test.go

```diff
@@ -0,0 +1,139 @@
+package charts
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/charts"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeconfig"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type WebhookTestSuite struct {
+	suite.Suite
+	client       *rancher.Client
+	session      *session.Session
+	clusterList  string
+	chartVersion string
+}
+
+func (w *WebhookTestSuite) TearDownSuite() {
+	w.session.Cleanup()
+}
+
+func (w *WebhookTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	w.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(w.T(), err)
+
+	w.client = client
+
+	// Get clusterName from config yaml
+	w.clusterList = client.RancherConfig.ClusterName
+	w.chartVersion, err = client.Catalog.GetLatestChartVersion(charts.RancherWebhookName)
+	require.NoError(w.T(), err)
+
+}
+
+func (w *WebhookTestSuite) ValidateEscalationCheck(client *rancher.Client) {
```

Comment:
Could you explain what is this usecase? Also can we add the details of the tests in the original PR, so its easier to review?

Reply:
This is to make sure we are checking the webhook functionality is working as expected. Details were added in the ticket here https://github.com/rancher/qa-tasks/issues/785. could you clarify what original PR is? do you mean description of the PR? 

- Create a Restricted Admin user
- Log in as the restricted admin user
- Attempt to edit the admin role and make it the default for new users (may need to use the old UI to do this)
- The attempt should fail, and display an error message indicating that the user is attempting to grant permissions that they don’t currently have